cimr/processor/tad.py

```diff
@@ -27,47 +32,150 @@ class Tadpole:
         chrom: chromosome id in the format of chr+number
         start: starting position of the tad block
         end: ending position of the tad block
-        (optional) id: unique id of the tad block
-        (optional) comment: optional column containing additional information
 
         e.g.
-        chr1	1960001	2400000	tad0|hg19|chr1:1960001-2400000	1000
+        chr1	1960001	2400000
+    
+    Following columns can be provided with appropriate cimr arguements
 
-    cell_type: cell or tissue type wherein TAD information was measured
-    species: species, default is 9606 (homo sapiens; human)
+        study_id: unique ID of the study, preferably a GEO unique identifier
+        pub_id: DOI of the article published / preprinted using the data
+                this ID will be listed as a part of the list of recommended 
+                citations when cimr-d is used for any other studies
+        species: species. Default is homo_sapiens (9606, human)
+        cell_type: cell or tissue type wherein TAD information was measured
+        data_type: data type. Default is tad.
 
     Notes
     =====
 
     """
 
-    HEADERS = ['chrom', 'start', 'end', 'id', 'comment', 'species', 'cell_type', 'pmid']
-
-    def __init__(self, file_name, species, cell_type):
+    HEADERS = [
+        'chrom', 
+        'start', 
+        'end', 
+        'feature_id'
+    ]
+
+    METAHEADERS = [
+        'file_name', 
+        'study_id', 
+        'pub_id', 
+        'species', 
+        'cell_type', 
+        'data_type', 
+        'entry_count',
+        'processed_date'
+    ]
+
+    METADATA_URL = 'https://raw.githubusercontent.com/greenelab/cimr-d/master/metadata/'
+
+
+    def __init__(self, 
+                 file_name, 
+                 study_id, 
+                 pub_id, 
+                 species, 
+                 cell_type, 
+                 data_type, 
+                 ):
         self.file_name = file_name
         self.template = pandas.DataFrame(columns=self.HEADERS)
+        self.study_id = study_id
+        self.pub_id = pub_id
         self.species = species
         self.cell_type = cell_type
+        self.data_type = data_type
+        self.processed_date = datetime.date.today().strftime('%Y-%m-%d')
+        self.metafile = self.METADATA_URL + data_type + '.txt'
+    
+
+    def open_metadata(self):
+        """Load or create a metadata for the new submission."""
+        try:
+            self.metadata = pandas.read_csv(self.metafile, sep='\t', header=0)
```

Comment:
Couldn't you change this to read table? Will the metadata file contain just tab separated list?

Reply:
I could. Not sure if there's necessarily an advantage either way since it is a small file atm and cimr dependency already includes pandas. It is a great point though. I made a related issue to consider this further https://github.com/greenelab/cimr/issues/20.